test/example/generated/ExampleDialect.cpp.inc

```diff
@@ -1671,8 +1670,10 @@ data
       void WriteVarArgOp::setData(::llvm::Value * data) {
         setArgOperand(0, data);
       }
-      ::llvm::iterator_range<::llvm::Value *> WriteVarArgOp::getArgs() {
-        return ::llvm::make_range((arg_begin() + 1)->get(), (arg_begin() + arg_size() - 1)->get());
```

Comment:
BTW, apart from the memory layout issue: Wouldn't the old version have missed the last element, passing it as `end`?

Reply:
No, otherwise the tests would have failed.

Reply:
Yes, that was also wrong. The tests just didn't capture that, which is why I changed the visitor test.

Reply:
What is the reason the test in test-builder.test did not fail then? Is this now working because of your change?

Reply:
Well, the code in ExampleMain.cpp was also wrong. It directly used the `end()` of the range, which should have failed because the end is supposed to be exclusive / one-past the range. But the bug in how `end` was set and the bug in how `end` was used canceled each other out.